lib/screen/main/code_scanner.dart

```diff
@@ -0,0 +1,106 @@
+import 'package:flutter/material.dart';
+import 'package:showcase_flutter_build_web_desktop/extension/context_extension.dart';
+import 'package:showcase_flutter_build_web_desktop/theme/dimen.dart';
+
+import '../../gen/assets.gen.dart';
+import '../../gen/colors.gen.dart';
+
+class CodeScanner extends StatelessWidget {
+  final VoidCallback _onClickScan;
+  final VoidCallback _onClickApply;
+  final String _placeHolderString;
+
+  const CodeScanner(
+      {Key? key,
+      required VoidCallback onClickScan,
+      required VoidCallback onClickApply,
+      required String placeHolderString})
+      : _onClickScan = onClickScan,
+        _onClickApply = onClickApply,
+        _placeHolderString = placeHolderString,
+        super(key: key);
```

Comment:
Optional: We can let them public for this case.

```suggestion
  final VoidCallback onClickScan;
  final VoidCallback onClickApply;
  final String placeHolderString;

  const CodeScanner({
    Key? key,
    required this.onClickScan,
    required this.onClickApply,
    required this.placeHolderString,
  }) : super(key: key);
```